src/StoreKeeper/WooCommerce/B2C/Imports/ProductImport.php

```diff
@@ -1033,29 +1042,65 @@ protected function updateProductMeta($newProduct, Dot $dotObject, array $log_dat
         $this->debug('storekeeper_id added to post.', $log_data);
 
         if ($dotObject->has('flat_product.content_vars')) {
+
             $nonAttributeOptions = array_filter(
                 $dotObject->get('flat_product.content_vars'),
                 function ($attribute) {
                     return !key_exists('attribute_option_id', $attribute);
                 }
             );
-            foreach ($nonAttributeOptions as $attribute) {
+
+            if ($dotObject->has('flat_product.product.has_addons')) {
+                update_post_meta(
+                    $newProduct->get_id(),
+                    self::META_HAS_ADDONS,
+                    $dotObject->get('flat_product.product.has_addons') ? '1' : '0',
+                );
+            }
+
+            // Find attributes that have 'attribute_option_color_hex'
+            $colorAttributes = array_filter(
+                $dotObject->get('flat_product.content_vars'),
+                function ($attribute) {
+                    return array_key_exists('attribute_option_color_hex', $attribute);
+                }
+            );
+
+            // Merge the non-attribute options with the color attributes
+            $mergedAttributes = array_merge($nonAttributeOptions, $colorAttributes);
+
+
+            foreach ($mergedAttributes as $attribute) {
                 if (key_exists('attribute_id', $attribute)) {
                     $label = sanitize_title($attribute['label']);
                     $attribute_id = $attribute['attribute_id'];
                     update_post_meta($newProduct->get_id(), 'attribute_id_'.$attribute_id, $label);
+                    // Check if 'attribute_option_color_hex' exists and save it
+                    if (key_exists('attribute_option_color_hex', $attribute)) {
+                        $color_hex = $attribute['attribute_option_color_hex'];
+                        // Check if the term already exists
+                        $value_label = $attribute['value_label']; // e.g., 'Orange'
+                        $term = term_exists($value_label, 'pa_color');
```

Comment:
you cannot hardcode the attribute name there can be many of them with different names